quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/transaction/TransactionalSessionHandler.java

```diff
@@ -49,22 +49,20 @@ default KeycloakSession create() {
      *
      * @param session a transactional session
      */
-    default void close(KeycloakSession session) {
-        if (DefaultKeycloakSession.class.cast(session).isClosed()) {
-            return;
-        }
+    default void close() {
+        RoutingContext context = ResteasyContext.getContextData(RoutingContext.class);
```

Comment:
Can we just get the session as an argument? I don't think we need to resolve contextual data if the expected instance can be easily passed to the method.

Reply:
Sadly, no: the session could be attempted to be closed twice that way. This led to duplicate closing of the same session.

Reply:
I see.

I do not think we should be adding more method calls if we can just avoid it by checking if the session is closed. As I've proposed before. Looks like a nitpick but I'm really trying to avoid unnecessary logic when processing the request lifecycle.

I know we have some divergences about adding the  `isClosed` method. So I would leave it to others to comment here.

Reply:
I don't like relying on the context-data to be cleared as a way to "check if the session has been closed". I think what @pedroigor is suggesting would be nicer, and having a `isClosed` method on the session makes sense to me.

Reply:
The session is created by the filter/interceptor/handler chain and should be closed in there.
Thus it is the handler chain which owns the session, an it maintains its state in the context data.
The session itself is not aware of when it should be closed.
Hence it seems to me that it should be the handler chain which should hold the state of the session, and to precisely define when it should be closed, and call `close` accordingly.
Relying on the active/closed state in the session to determine whether it should be closed object seems like violation of the encapsulation to me.

If this should be changed to a `isClosed` method regardless, I can do that. Let me know your thoughts.

Reply:
This part will be removed in the next round